prow/config/config.go

```diff
@@ -535,6 +542,26 @@ type Plank struct {
 	JobURLPrefixDisableAppendStorageProvider bool `json:"jobURLPrefixDisableAppendStorageProvider,omitempty"`
 }
 
+type ProwJobDefaultEntry struct {
+	// Matching/filtering fields. All filters must match for an entry to match.
+
+	// OrgRepo matches against the "org" or "org/repo" that the presubmit or postsubmit
+	// is associated with. If the job is a periodic, extra_refs[0] is used. If the
+	// job is a periodic without extra_refs, the empty string will be used.
+	// If this field is omitted all jobs will match.
+	OrgRepo string `json:"repo,omitempty"`
+	// Cluster matches against the cluster alias of the build cluster that the
+	// ProwJob is configured to run on. Recall that ProwJobs default to running on
+	// the "default" build cluster if they omit the "cluster" field in config.
+	Cluster string `json:"cluster,omitempty"`
```

Comment:
nit: Since the matching fields are the same between `ProwJobDefaultEntry` and `DefaultDecorationConfigEntry` we could stick them into their own struct and make that an embedded struct of them both. That would allow the `matches(...)` function to become a method of the matcher struct and the two existing `matches(...)` methods could be omitted.

This is a minor change and there might be something I'm overlooking that makes you implementation better. Just something to consider to make this a bit more idiomatic.

Reply:
I think I might misunderstand how embedding works. My understanding is that it would make the current uses of DefaultDecorationConfig not backwards compatible because they would need to define `DefaultDecorationConfigEntry{ Matcher: { OrgRepo... } }` instead of `DefaultDecorationConfigEntry{ OrgRepo: ... }` Am I missing something?

Reply:
Here is an example in the go playground: https://play.golang.org/p/4J79nWGT21L
It will be compatible with existing config, but I think existing definitions in code would need to be updated. It doesn't make a difference to users to if that is a pain to switch then don't worry about it.

Reply:
I think this is the correct thing to do, but this PR is blocking other work so I left a todo to clean it up later.